website/docs/guides/resources.md

```diff
@@ -0,0 +1,215 @@
+1. We want to be able to interact with them while they are running
+
+## Why resources?
+
+As an example, let's consider that our program opens a Socket, and we want to be
+able to send messages to this socket while it is open. This is fairly simple
+to write using regular operations like this:
+
+``` javascript
+import { main } from 'effection';
+import { Socket } from 'net';
+
+main(function*() {
+  let socket = new Socket();
+  socket.connect(1337, '127.0.0.1');
+
+  yield once(socket, 'connect');
+
+  socket.write('hello');
+
+  socket.close();
+});
+```
+
+This works, but there are a lot of details we need to remember to use the socket
+safely. It would be nice if we could create a friendly abstraction that we could
+reuse in all of our code that uses socket.
+
+We know we want to close this socket once we're done, so our first attempt might
+look something like this:
+
+``` javascript
+import { once } from 'effection';
+
+export function *createSocket(port, host) {
+  let socket = new Socket();
+  socket.connect(port, host);
+
+  yield once(socket, 'connect');
+
+  try {
+    yield
+    return socket;
+  } finally {
+    socket.close();
+  }
+});
+```
+
+But when we actually try to use our `createSocket` operation, we run into a problem:
+
+``` javascript
+import { main } from 'effection';
+import { createSocket } from './create-socket';
+
+main(function*() {
+  let socket = yield createSocket(1337, '127.0.0.1'); // this blocks forever
+  socket.write('hello'); // we never get here
+});
+```
+
+## Using resources
+
+Remember our criteria from before:
+
+1. Socket is a long running process
+1. We want to interact with the socket while it is running by sending messages to it
+
+This is a good use-case for using a Resource. Let's look at how we can rewrite
+`createSocket` using resources, an implementation might look something like
+this:
+
+``` javascript
+import { once, spawn } from 'effection';
+
+export function createSocket(port, host) {
+  return {
+    *init() {
+      let socket = new Socket();
+      socket.connect(port, host);
+
+      yield spawn(function* closeSocket() {
+        try {
+          yield
+        } finally {
+          socket.close();
+        }
+      });
+
+      yield once(socket, 'connect');
+
+      return socket;
+    }
+  }
+});
+```
+
+Before we unpack what's going on, let's just note that how we use `createSocket` has
+not changed at all, only it now works as we expect!
+
+``` javascript
+import { main } from 'effection';
+import { createSocket } from './create-socket';
+
+main(function*() {
+  let socket = yield createSocket(1337, '127.0.0.1'); // waits for the socket to connect
+  socket.write('hello'); // this works
+  // once `main` finishes, the socket is closed
+});
+```
+
+## How resources work
+
+Let's take a look at the implementation of `createSocket`. Effection considers
+any object which has an `init` function a Resource, and so the object returned
+from `createSocket` is a Resource.
```

Comment:
I think we should mention this _before_ the example on line 88.

Reply:
Good call, I've rewritten this slightly